src/Microsoft.Data/DataFrameBuffer.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Data
+{
+    public class DataFrameBuffer<T>
+        where T : struct
+    {
+        // TODO: Change this to Memory<T>
+        public Memory<byte> Memory { get; private set; }
+        private readonly int _size;
+
+        private int Capacity => Memory.Length / _size;
+
+        public int MaxCapacity => Int32.MaxValue / _size;
+
+        public Span<T> Span
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Generally we try to avoid inlining attributes unless we know for sure we need them. ie., we know it will not get inlined (for some bad reason) and we know inlining is important to get the performance we want. In this case I would trust it will get inlined, and when we do perf work later, if it's not, we can do appropriate things (which might include modifying the jit)

Reply:
Got it. This will go away soon enough though for https://github.com/dotnet/corefxlab/issues/2659